ast/desugar/Desugar.cc

```diff
@@ -1278,20 +1278,16 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                 result = std::move(res);
             },
             [&](parser::IRange *ret) {
-                TreePtr range = MK::Constant(loc, core::Symbols::Range());
                 auto from = node2TreeImpl(dctx, std::move(ret->from));
                 auto to = node2TreeImpl(dctx, std::move(ret->to));
-                auto send = MK::Send2(loc, std::move(range), core::Names::new_(), std::move(from), std::move(to));
-                result = std::move(send);
+                auto range = MK::Range(loc, std::move(from), std::move(to), false);
+                result = std::move(range);
```

Comment:
Here's where it should construct the call to the buildRange intrinsic, instead of constructing the Range node. We do something similar with literals for arrays, hashes, and string interpolation.

https://github.com/sorbet/sorbet/blob/33a668e7238aaab1f4164cba4c358b8dd529b5b3/ast/desugar/Desugar.cc#L144-L145